test/test_utils.py

```diff
@@ -82,12 +82,22 @@ def assert_expected(actual: Any, expected: Any, rtol: float = None, atol: float
     )
 
 
+def tuple_to_dict(t):
+    if not isinstance(t, tuple):
+        raise TypeError(f"Input must be of type tuple but got {type(t)}")
+
+    return {k: v for k, v in enumerate(t)}
+
+
+def is_named_tuple(nt):
+    # hacky way to assert an instance of NamedTuple
+    return isinstance(nt, tuple) and hasattr(nt, "_asdict")
```

Comment:
I notice in a [similar question](https://stackoverflow.com/questions/2166818/how-to-check-if-an-object-is-an-instance-of-a-namedtuple) on stack overflow they also check `hasattr(nt, "_fields")`. Any reason we're not doing that here?

Reply:
`NamedTuple` has a couple of attributes where `_asdict` is what matter in this application. If I were to assert for `_field` only it's just another attr to assert.

```
>>> class nt(NamedTuple):
...     x: int
...     y: str
...
>>> a = nt(x=1, y=2)
>>> dir(a)
['__add__', '__annotations__', '__class__', '__contains__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__getnewargs__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__module__', '__mul__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__rmul__', '__setattr__', '__sizeof__', '__slots__', '__str__', '__subclasshook__', '_asdict', '_field_defaults', '_field_types', '_fields', '_fields_defaults', '_make', '_replace', 'count', 'index', 'x', 'y']
```